### PR TITLE
FIX : wrong tests on fetch

### PR DESCRIPTION
**Pull request by atm-gauthier**

### CODE REVIEW

File: htdocs/user/class/user.class.php [hunk 1]
```diff
@@ -1483,9 +1483,9 @@ function update($user, $notrigger=0, $nosyncmember=0, $nosyncmemberpass=0, $nosy
 					// This user is linked with a member, so we also update member information
 					// if this is an update.
 					$adh=new Adherent($this->db);
-					$result=$adh->fetch($this->fk_member);
+					$result=$adh->fetch('',$this->fk_member);
```
Comment (eldy) [inline htdocs/user/class/user.class.php]: Do you mean you have a user into an entity and its linked member into another one ?

this->fk_member is a int key.
So the fetch must be done in the first parameter (the primary key).

What it is possible to do, is to test $adh->entity after the fetch, and if it is not on same entity than $user->entity, we ignore the update of the member.

Comment (atm-gauthier) [reply htdocs/user/class/user.class.php]: @eldy yes, it is the problem.
i'm writting new correction

File: htdocs/user/class/user.class.php [hunk 2]
```diff
@@ -1483,9 +1483,9 @@ function update($user, $notrigger=0, $nosyncmember=0, $nosyncmemberpass=0, $nosy
 					// This user is linked with a member, so we also update member information
 					// if this is an update.
 					$adh=new Adherent($this->db);
-					$result=$adh->fetch('',$this->fk_member);
+					$result=$adh->fetch($this->fk_member);
 
-					if ($result > 0)
+					if ($result > 0 && $adh->entity == $conf->entity)
```
Comment (hregis) [inline htdocs/user/class/user.class.php]: @atm-gauthier @eldy 
i don't understand why ??
what happens if we use the centralization of users on the main entity with a sharing of the membership base?

Comment (atm-gauthier) [reply htdocs/user/class/user.class.php]: @hregis you're right sorry, problem i had was with tests on fetch functions

**Pull request merged**